Tasks/MavenV3/maventask.ts

```diff
@@ -565,4 +568,20 @@ function execBuildWithRestore() {
         execBuild();
     }
 }
+
+function replaceGifToBase64() {
+    try {
+        const htmlString = fs.readFileSync(path.join(reportDirectory, 'index.html'), 'utf-8');
+        const dom = new JSDOM(htmlString);
```

Comment:
Let's try to avoid procedure approach and consider KISS principle - this method does a lot things (reads html file, deserializes it, replaces image and writes changes back to the file). The better option would probably be to decompose this into several methods.

Reply:
Split to 3 methods.